rule.go

```diff
@@ -77,32 +81,91 @@ func (cc *Conn) GetRule(t *Table, c *Chain) ([]*Rule, error) {
 	return rules, nil
 }
 
-// AddRule adds the specified Rule. See also
-// https://wiki.nftables.org/wiki-nftables/index.php/Simple_rule_management
-func (cc *Conn) AddRule(r *Rule) *Rule {
+// GetRuleHandle returns a specific rule's handle. Rule is identified by Table, Chain and RuleID.
+func (cc *Conn) GetRuleHandle(t *Table, c *Chain, ruleID uint32) (uint64, error) {
+	conn, err := cc.dialNetlink()
+	if err != nil {
+		return 0, err
+	}
+	defer conn.Close()
+	if ruleID == 0 {
+		return 0, fmt.Errorf("Rule's RuleID cannot be 0")
+	}
+
+	data, err := netlink.MarshalAttributes([]netlink.Attribute{
+		{Type: unix.NFTA_RULE_TABLE, Data: []byte(t.Name + "\x00")},
+		{Type: unix.NFTA_RULE_CHAIN, Data: []byte(c.Name + "\x00")},
+		{Type: unix.NFTA_RULE_USERDATA, Data: binaryutil.BigEndian.PutUint32(ruleID)},
+	})
+	if err != nil {
+		return 0, err
+	}
+	message := netlink.Message{
+		Header: netlink.Header{
+			Type:  netlink.HeaderType((unix.NFNL_SUBSYS_NFTABLES << 8) | unix.NFT_MSG_GETRULE),
+			Flags: netlink.Request | netlink.Acknowledge | netlink.Dump | unix.NLM_F_ECHO,
+		},
+		Data: append(extraHeader(uint8(t.Family), 0), data...),
+	}
+	if _, err := conn.SendMessages([]netlink.Message{message}); err != nil {
+		return 0, fmt.Errorf("SendMessages: %v", err)
+	}
+	reply, err := conn.Receive()
+	if err != nil {
+		return 0, fmt.Errorf("Receive: %v", err)
+	}
+	if len(reply) != 1 {
+		return 0, fmt.Errorf("Receive: Expected 1 message but got %d", len(reply))
+	}
+	rr, err := ruleFromMsg(reply[0])
+	if err != nil {
+		return 0, err
+	}
+
+	return rr.Handle, nil
+}
+
+// AddRule adds the specified Rule
+func (cc *Conn) AddRule(r *Rule) (*Rule, error) {
```

Comment:
Can you revert the change to the function signature please? You’re only ever returning nil AFAICT?

Reply:
Done